core/validation/initial_check.py

```diff
@@ -13,7 +13,13 @@
 def extract_submission_details(
     workbook: dict[str, pd.DataFrame],
     reporting_round: int,
-) -> dict[str, list[str]] | dict[str, dict[Any, Any]]:
+    place_names: tuple[str] | None,
+) -> (
+    dict[str, list[str]]
+    | dict[str, list[str]]
+    | dict[str, tuple[Any, set[Any]]]
+    | dict[str, tuple[Any, set[Any]] | tuple[Any, set[str]]]
```

Comment:
This feels a bit confusing and not very useful - can it be simplified?

Reply:
This is what pycharm updated the returns to.

I think:
```
dict[str, tuple[str]]
```
captures all of the use cases apart from MissingSheets and InvalidSheets, which we don't expect the user to hit anyway.

Reply:
That looks more useful. This probably highlights that this function is doing too many things but not something to change as part of this PR.